sympy/calculus/util.py

```diff
@@ -1597,3 +1597,155 @@ def union(self, other):
+    function is not differentiable. Also, the domain should be a subset of
+    S.Reals
+
+    Parameters
+    ==========
+
+    f: expr
+        The concerned expression
+    symbol: Symbol
+        Variable for which the arguments to be determined.
+    domain: Set
+        The domain over which the arguments are to be determined
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, sin, Interval, S, exp, pi
+    >>> from sympy.calculus.util import argmax_real
+
+    >>> x = Symbol('x')
+    >>> argmax_real(sin(x), x, Interval(-2*pi, pi))
+    FiniteSet(-3*pi/2, pi/2)
+
+    >>> argmax_real(-x**2 + 5*x - 6)
+    FiniteSet(5/2)
+
+    >>> argmax_real(exp(x), x, S.Reals)
+    EmptySet
+    """
+    f = _sympify(f)
+    if symbol is None:
+        if len(f.free_symbols) > 1:
+            raise ValueError(filldedent('''
+                Variable for which argmax is to be determined needs to be
+                specified.'''))
+        else:
+            symbol = list(f.free_symbols)[0]
+
+    if symbol not in f.free_symbols:
+        raise ValueError(filldedent('''
+            argmax of %s cannot be calculated with respect to %s'''%
+            (f, symbol)))
+
+    return _argMaxMin(f, symbol , domain, max = True)
+
+def argmin_real(f, symbol = None, domain = S.Reals):
+    """
+    Returns the set of arguments of minima, i.e., values of `symbol`
+    for which the `f` attains the minimum value in the given domain.
+    Note: This function uses derivatives to calculate the points where the
+    minimum value is achieved, and thus, would not include points where the
+    function is not differentiable. Also, the domain should be a subset of
+    S.Reals
+
+    Parameters
+    ==========
+
+    f: expr
+        The concerned expression
+    symbol: Symbol
+        Variable for which the arguments to be determined.
+    domain: Set
+        The domain over which the arguments are to be determined
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, sin, Interval, S, exp, pi
+    >>> from sympy.calculus.util import argmin_real
+
+    >>> x = Symbol('x')
+    >>> argmin_real(sin(x), x, Interval(-2*pi, pi))
+    FiniteSet(-pi/2)
+
+    >>> argmin_real((x - 1)**2, x, S.Reals)
+    FiniteSet(1)
+
+    >>> argmin_real(exp(x), x, S.Reals)
+    EmptySet
+    """
+    f = _sympify(f)
+    if symbol is None:
+        if len(f.free_symbols) > 1:
+            raise ValueError(filldedent('''
+                Variable for which argmin is to be determined needs to be
+                specified.'''))
+        else:
+            symbol = list(f.free_symbols)[0]
+    if symbol not in f.free_symbols:
+        raise ValueError(filldedent('''
+            argmax of %s cannot be calculated with respect to %s'''%
+            (f, symbol)))
+
+    return _argMaxMin(f, symbol , domain, max = False)
+
+def _argMaxMin(f, symbol, domain, max):
+    """
+    Helper function for calculuating argmin/argmax
```

Comment:
Should these maybe be `argmin_real` and `argmax_real`?

Reply:
Well @asmeurer said [here](https://github.com/sympy/sympy/pull/18550#issuecomment-582744068) that we will add support for non-real domains in future so the names can be ```argmax``` and ```argmin```.